### PR TITLE
Fix k0s status components

### PR DESCRIPTION
**Pull request by juanluisvaladas**
## Description

If a component.Healthy returns an error which is not nil the json would be marshalled and unmarshalled incorrectly. Since the error is only for humans we're storing it as a string.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## How Has This Been Tested?

- [x] Manual test
- [ ] Auto test added

## Checklist:

- [x] My code follows the style [guidelines](https://github.com/k0sproject/k0s/blob/main/docs/contributors/overview.md) of this project 
- [x] My commit messages are [signed-off](https://github.com/k0sproject/k0s/blob/main/docs/contributors/github_workflow.md)
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Review by mikhail-sakhnov — APPROVED** [review]

File: pkg/component/prober/prober.go [hunk 1]
```diff
@@ -161,10 +161,15 @@ func (p *Prober) checkComponentsHealth(ctx context.Context, at time.Time) {
 			p.healthCheckState[name] = ring.New(p.probesTrackLength)
 		}
 		// TODO: add back-off logic
+		err := ""
```
Comment (kke) [inline pkg/component/prober/prober.go:164]: Having something other than an error in `err` is a bit confusing.

File: pkg/component/prober/prober.go [hunk 2]
```diff
@@ -215,5 +220,5 @@ func (p *Prober) Register(name string, component any) {
 type ProbeResult struct {
 	Component string    `json:"component"`
 	At        time.Time `json:"at"`
-	Error     error     `json:"error"`
+	Error     string    `json:"error"`
```
Comment (kke) [inline pkg/component/prober/prober.go:223]: Maybe it needs omitempty if it isn't error interface.

File: pkg/component/prober/prober.go [hunk 3]
```diff
@@ -161,10 +161,15 @@ func (p *Prober) checkComponentsHealth(ctx context.Context, at time.Time) {
 			p.healthCheckState[name] = ring.New(p.probesTrackLength)
 		}
 		// TODO: add back-off logic
+		err := ""
+		healthy := component.Healthy()
```
Comment (kke) [inline pkg/component/prober/prober.go:165]: This should be the the `err` :)

**Comment by kke** [comment]
Could also be a custom MarshalJSON on ProbeResult:

```go
func (pr *ProbeResult) MarshalJSON() ([]byte, error) {
  type probeResult ProbeResult
  
  var errStr *string
  if pr.Error != nil {
    errStr = &pr.Error.Error()
  }
  
  return json.Marshal(&struct {
    *probeResult
    Error *string `json:"error"`
  }{
    probeResult: (*probeResult)(pr),
    Error:      errStr,
  })
}
```

Is this the only `Healthy()` implementation or am I not looking hard enough? 🤔 

```go
func (k *Konnectivity) Healthy() error {
	if k.clusterConfig == nil {
		return fmt.Errorf("cluster config not yet available")
	}

	return nil
}
```

File: pkg/component/prober/prober.go (hunk 3 above)
Comment (juanluisvaladas) [reply pkg/component/prober/prober.go:165]: true

**Comment by juanluisvaladas** [comment]
> Could also be a custom MarshalJSON on ProbeResult:
That's probably the sanest option we'll need both custom marshal and unmarshal. I'll do that in the plane.

> Is this the only `Healthy()` implementation or am I not looking hard enough? 🤔

It is at the moment. I have it implemented in my local branch for etcd as well but when there is an error it doesn't print it.

**Comment by juanluisvaladas** [comment]
Disregard this change, yesterday it seemed cleaner than doing the custom marshalling and unmarshalling but it's not

**Review by kke — APPROVED** [review]

**Pull request merged**